x-pack/plugins/search_indices/public/components/connection_details/connection_details.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiButtonIcon,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../hooks/use_kibana';
+
+const DEFAULT_ELASTICSEARCH_URL = 'http://localhost:9200';
+
+export const ConnectionDetails: React.FC = () => {
+  const { euiTheme } = useEuiTheme();
+  const { cloud } = useKibana().services;
+
+  const elasticsearchUrl = cloud?.elasticsearchUrl ?? DEFAULT_ELASTICSEARCH_URL;
```

Comment:
we might need to introduce a hook for this, I know @sphilipse is working on a PR that will support reading the public elasticsearch URL, but it has to be read in the server.